vault/logical_system_integ_test.go

```diff
@@ -31,219 +31,306 @@ const (
 	expectedEnvValue = "BAR"
 )
 
-func TestSystemBackend_Plugin_secret(t *testing.T) {
-	cluster := testSystemBackendMock(t, 1, 1, logical.TypeLogical)
-	defer cluster.Cleanup()
+// logicalVersionMap is a map of version to test plugin
+var logicalVersionMap = map[string]string{
+	"v4": "TestBackend_PluginMain_V4_Logical",
+	"v5": "TestBackend_PluginMainLogical",
+}
 
-	core := cluster.Cores[0]
+// credentialVersionMap is a map of version to test plugin
+var credentialVersionMap = map[string]string{
+	"v4": "TestBackend_PluginMain_V4_Credentials",
+	"v5": "TestBackend_PluginMainCredentials",
+}
 
-	// Make a request to lazy load the plugin
-	req := logical.TestRequest(t, logical.ReadOperation, "mock-0/internal")
-	req.ClientToken = core.Client.Token()
-	resp, err := core.HandleRequest(namespace.RootContext(nil), req)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
-	if resp == nil {
-		t.Fatalf("bad: response should not be nil")
+func TestSystemBackend_Plugin_secret(t *testing.T) {
+	testCases := []struct {
```

Comment:
this probably could have just been any array of strings but maybe we want to add to this struct in the future?

Reply:
I'm good with using the struct. Common pattern and makes it easy to add to in the future 👍